### PR TITLE
Add Spell Selection Logic

### PR DESCRIPTION
**Pull request by daniela-venuta**
Add logic to select a spell and set necessary values on the start of a run. Still needs to be integrated with the appropriate menu

### CODE REVIEW

File: Hush/Assets/Scripts/Game/SpellSelectorManager.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            GameManager.Instance.SetHeavySpellCooldownTime(spell.HeavyCooldown);
+        }
+
+        private ISpell GetSpellComponent(SpellType type)
+        {
+            ISpell spell;
+            switch (type)
+            {
+                case SpellType.FireballSpell:
+                    spell = player.GetComponent<FireballSpell>();
```
Comment (po-trottier) [inline Hush/Assets/Scripts/Game/SpellSelectorManager.cs:47]: You don't need this you can just GetComponent<ISpell> directly and it's going to return you whichever component matches

File: Hush/Assets/Scripts/Weapon/InvisibleSpell.cs [hunk 2]
```diff
@@ -18,15 +18,12 @@ public class InvisibleSpell : MonoBehaviour, ISpell
...
-        //TODO Remove
-        private void Awake()
-        {
-            // GameManager.Instance.SetActiveHeavySpell(SpellType.InvisibleSpell);
-            // GameManager.Instance.SetActiveLightSpell(SpellType.InvisibleSpell);
-        }
+        public SpellType SpellType => SpellType.InvisibleSpell;
+        
+        public float HeavyCooldown => 10f;
+        public float LightCooldown => 6f;
```
Comment (po-trottier) [inline Hush/Assets/Scripts/Weapon/InvisibleSpell.cs:26]: Probably want those as SerializeField for all spells

Comment (po-trottier) [reply Hush/Assets/Scripts/Weapon/InvisibleSpell.cs:26]: [SerializeField] private float heavyCooldown = 10f;

public float HeavyCooldown => heavyCooldown;

File: Hush/Assets/Scripts/Game/SpellSelectorManager.cs (hunk 1 above)
Comment (daniela-venuta) [reply Hush/Assets/Scripts/Game/SpellSelectorManager.cs:47]: Right now we have all of the Spells attached as Components to the player, so unless we detach and reattach them through this, wont there be confusion over which one it needs if I do something like GetComponent<ISpell> ? Not sure if this is what you mean?

Comment (po-trottier) [reply Hush/Assets/Scripts/Game/SpellSelectorManager.cs:47]: Makes sense! I guess that's okay then!

File: Hush/Assets/Scripts/Game/SpellSelectorManager.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+using UnityEngine;
+using Weapon;
+using Weapon.Enums;
+
+namespace Game
+{
+    public class SpellSelectorManager : MonoBehaviour
+    {
+        public static SpellSelectorManager Instance;
+        [SerializeField] private GameObject player;
```
Comment (po-trottier) [inline Hush/Assets/Scripts/Game/SpellSelectorManager.cs:11]: If you want to be consistent with the rest of the Managers get the player via Tag or FindObjectOfType

It juste makes the managers easier to use, never need to set references in the inspector

**Review by po-trottier — DISMISSED** [review]
Minor change if you want, otherwise good

**Review by emilie-martin — DISMISSED** [review]
looks good to me! can't wait to have the UI implemented for this; it'll look great
(but also, you seem to have a merge conflict in the dev scene)

**Review by arejayelle — APPROVED** [review]
Logic-wise seems fine.
Can't really test how it's going to be integrated, but I guess that's being left for another task.
I agree w/ P-O on the manager architecture stuff but honestly that's pretty minor.

**Review by arejayelle — DISMISSED** [review]
LGTM

**Review by po-trottier — DISMISSED** [review]

**Review by po-trottier — APPROVED** [review]

**Pull request merged**